scripts/Dockerfile.e2e

```diff
@@ -0,0 +1,76 @@
+# syntax=docker/dockerfile:1.7-labs
+# (use non-stable syntax for convenient --parents option in COPY command)
+
+# Download clang
+FROM scratch AS clang
+ADD https://github.com/llvm/llvm-project/releases/download/llvmorg-16.0.4/clang+llvm-16.0.4-x86_64-linux-gnu-ubuntu-22.04.tar.xz /clang.tar.xz
+
+FROM ubuntu:latest
```

Comment:
Is using ubuntu:latest a good idea? It feels like this could be brittle and could be better done by pinning a specific build here like ubuntu:22.04 or ubuntu:24.04